src/SentryFullStory.ts

```diff
@@ -23,34 +25,43 @@ class SentryFullStory {
   baseSentryUrl: string;
   client: FullStoryClient;
 
-  constructor(sentryOrg: string, options: Options = {}) {
+  constructor(sentryOrg: string, options: Options) {
     this.sentryOrg = sentryOrg;
-    this.client = options.client || FullStory;
+    this.client = options.client;
     this.baseSentryUrl = options.baseSentryUrl || 'https://sentry.io';
   }
 
   setupOnce() {
-    Sentry.addGlobalEventProcessor((event: Event, hint?: EventHint) => {
-      const self = Sentry.getCurrentHub().getIntegration(SentryFullStory);
+    addGlobalEventProcessor(async (event, hint) => {
```

Comment:
You might not even need `@sentry/core` since `addGlobalEventProcessor` and `getCurrentHub` are dependency injected into `setupOnce`.

See: https://github.com/getsentry/sentry-javascript/blob/2d068e9f944a96dd566519a9e38ae05fc7b5b769/packages/types/src/integration.ts#L25